### PR TITLE
Add savedstates command

### PR DESCRIPTION
**Pull request by HumorousFool**
This adds a command for players to see a list of their own saved states and for admins to see any player's saved states.

This has been compiled and test and has proper internationalization.

I apologize for not opening an issue about this, it has been discussed many times on APDevTeam/Movecraft and I thought an issue for it was open here.

### CODE REVIEW

File: plugin/src/main/java/net/countercraft/movecraft/repair/commands/SavedStatesCommand.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+public class SavedStatesCommand implements CommandExecutor {
+
+    @Override
+    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
+        if (!command.getName().equalsIgnoreCase("savedstates")) {
+            return false;
+        }
+
+        if(!sender.hasPermission("movecraftrepair.savedstates")) {
```
Comment (TylerS1066) [inline plugin/src/main/java/net/countercraft/movecraft/repair/commands/SavedStatesCommand.java:24]: Let's use "movecraft.repair.command.savedstates"

File: plugin/src/main/java/net/countercraft/movecraft/repair/commands/SavedStatesCommand.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            return true;
+        }
+
+        OfflinePlayer player = null;
+        int page = 1;
+        if(args.length > 0) {
+            if(args.length == 1 && args[0].matches("-?\\d+")) {
+                page = Integer.parseInt(args[0]);
+            }
+            else if(sender.hasPermission("movecraftrepair.savedstates.other")) {
```
Comment (TylerS1066) [inline plugin/src/main/java/net/countercraft/movecraft/repair/commands/SavedStatesCommand.java:35]: Similar here

File: plugin/src/main/java/net/countercraft/movecraft/repair/commands/SavedStatesCommand.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            sender.sendMessage(MOVECRAFT_COMMAND_PREFIX + I18nSupport.getInternationalisedString("SavedStates - No Saved States"));
+            return true;
+        }
+
+
+        TopicPaginator paginator = new TopicPaginator(I18nSupport.getInternationalisedString("SavedStates - Saved States"));
+        for(File file : playerDirectory.listFiles()) {
+            if(file.isFile())
+            {
+                paginator.addLine(file.getName().replace(".schematic", ""));
```
Comment (TylerS1066) [inline plugin/src/main/java/net/countercraft/movecraft/repair/commands/SavedStatesCommand.java:69]: 1.13+ schematics are .schem iirc, which may mean we need to implement some sort of conversion process or update the saving name.

**Review by TylerS1066 — CHANGES_REQUESTED** [review]

**Comment by HumorousFool** [comment]
While .schem files are 1.13+ WE's default schematic files Movecraft-Repair still uses .schematic files so I've allowed it to handle both in case we switch. Otherwise, requested changes have been made.

**Comment by HumorousFool** [comment]
How does this look?

File: plugin/src/main/resources/plugin.yml [hunk 4]
```diff
@@ -5,4 +5,17 @@ api-version: 1.13
...
+commands:
+  savedstates:
+    description: Lists your saved repair states.
+    aliases: ["repairstates"]
+    usage: /<command> [Player] [Page]
+
+permissions:
+  movecraftrepair.command.savedstates:
+    description: Allows you to see a list of your saved states.
+    default: true
```
Comment (TylerS1066) [inline plugin/src/main/resources/plugin.yml:19]: This should be of the form `movecraft.repair.____`.  This should also default to op, let owners decide if they want to add the permission.

File: plugin/src/main/resources/plugin.yml [hunk 5]
```diff
@@ -5,4 +5,17 @@ api-version: 1.13
...
+  movecraftrepair.command.savedstates.other:
+    description: Allows you to see the saved states of other players.
```
Comment (TylerS1066) [inline plugin/src/main/resources/plugin.yml:21]: This should be of the form `movecraft.repair.____`.  This should also default to op.

**Review by TylerS1066 — COMMENTED** [review]
This looks pretty good, I just have a few notes.

I unfortunately do not have a Movecraft 8.0 test server setup yet, so I am unable to test and approve this PR yet.

**Comment by HumorousFool** [comment]
Requested changes made

**Review by TylerS1066 — APPROVED** [review]
Looks good!

It's going to be a while until I have a 1.14.4 server setup to test this, but if it works then I'll merge it.

**Comment by TylerS1066** [comment]
Just as a heads up, as of https://github.com/APDevTeam/Movecraft-Repair/commit/ad3da4616f14faf38678d5773e20b0ae1d5912d4 there have been some breaking changes to how Movecraft-Repair saves schematics (we now default to the proper .schem extension for sponge format schematics).  In the coming weeks, it's also planned to remove the 6.0 to 7.0 upgrade system and implement a .schematic (MCEdit format) to .schem conversion.

**Comment by HumorousFool** [comment]
Ok I'll update this when I can

**Pull request closed without merging**